### PR TITLE
Adds HTable.getRegionLocations(startKey, endKey) to FakeHTable.

### PR DESCRIPTION
**Pull request by daisyzhou**

### CODE REVIEW

**Comment by daisyzhou** [comment]
@kryzthov I need to add this method to FakeHTable for my unit test for KIJIMR-25 to work.

File: src/main/scala/org/kiji/testing/fakehtable/FakeHTable.scala [hunk 1]
```diff
@@ -688,6 +688,24 @@ class FakeHTable(
     return map
   }
 
+  def getRegionsInRange(startKey: Bytes, endKey: Bytes): JList[HRegionLocation] = {
+    val endKeyIsEndOfTable = Bytes.equals(endKey, HConstants.EMPTY_END_ROW);
```
Comment (kryzthov) [inline src/main/scala/org/kiji/testing/fakehtable/FakeHTable.scala]: nit: no semi-colon

File: src/main/scala/org/kiji/testing/fakehtable/FakeHTable.scala [hunk 2]
```diff
@@ -688,6 +688,24 @@ class FakeHTable(
     return map
   }
 
+  def getRegionsInRange(startKey: Bytes, endKey: Bytes): JList[HRegionLocation] = {
```
Comment (kryzthov) [inline src/main/scala/org/kiji/testing/fakehtable/FakeHTable.scala:694]: could you write some documentation?

**Comment by daisyzhou** [comment]
@kryzthov done, sorry.  I didn't take the time to check :(

File: src/main/scala/org/kiji/testing/fakehtable/FakeHTable.scala [hunk 3]
```diff
@@ -688,6 +688,25 @@ class FakeHTable(
     return map
   }
 
+  /** See HTable.getRegionsInRange(startKey, endKey). */
+  def getRegionsInRange(startKey: Bytes, endKey: Bytes): JList[HRegionLocation] = {
+    val endKeyIsEndOfTable = Bytes.equals(endKey, HConstants.EMPTY_END_ROW)
+    if ((Bytes.compareTo(startKey, endKey) > 0) && !endKeyIsEndOfTable) {
```
Comment (kryzthov) [inline src/main/scala/org/kiji/testing/fakehtable/FakeHTable.scala:696]: I believe this should be ≥ 0 : startKey is inclusive and endKey is exclusive

Comment (daisyzhou) [inline src/main/scala/org/kiji/testing/fakehtable/FakeHTable.scala:696]: Huh. Bug in HBase source? It's copied from there.

File: src/main/scala/org/kiji/testing/fakehtable/FakeHTable.scala [hunk 4]
```diff
@@ -688,6 +688,27 @@ class FakeHTable(
...
+    * Adapted from org.apache.hadoop.hbase.client.HTable.
+    */
+  def getRegionsInRange(startKey: Bytes, endKey: Bytes): JList[HRegionLocation] = {
+    val endKeyIsEndOfTable = Bytes.equals(endKey, HConstants.EMPTY_END_ROW)
+    if ((Bytes.compareTo(startKey, endKey) > 0) && !endKeyIsEndOfTable) {
+      throw new IllegalArgumentException(
+        "Invalid range: " + Bytes.toStringBinary(startKey) +
+        " > " + Bytes.toStringBinary(endKey))
+    }
+    val regionList = new JArrayList[HRegionLocation]();
```
Comment (kryzthov) [inline src/main/scala/org/kiji/testing/fakehtable/FakeHTable.scala:701]: other semi-colons, and below too

Comment (kryzthov) [inline src/main/scala/org/kiji/testing/fakehtable/FakeHTable.scala:701]: well, actually, don't bother, I'll fix and push!

Thanks for the change

**Comment by kryzthov** [comment]
b8f5e7b851eba2a7681bf5efc325c926e7a690f5

**Pull request closed without merging**